src/librustc/ty/layout.rs

```diff
@@ -816,12 +874,23 @@ impl<'a, 'gcx, 'tcx> Struct {
         }
         Ok(None)
     }
+
+    pub fn over_align(&self) -> Option<u32> {
+        let align = self.align.abi();
+        let abi_align = self.abi_align.abi();
+        if align > abi_align {
+            Some(align as u32)
+        } else {
+            None
+        }
+    }
 }
 
 /// An untagged union.
 #[derive(PartialEq, Eq, Hash, Debug)]
 pub struct Union {
     pub align: Align,
+    pub abi_align: Align,
```

Comment:
Same, although `field_align` might not work as well here. `member_align` might work for both.